### PR TITLE
Do not show false "Free shipping" text until it is real

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | This is common problem with PrestaShop, we always give people false statement at first that shipping is free even if this is not real |
| Type? | improvement |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | Probably |
| How to test? | "Free shipping" will only be showed up if it's true, if shipping is free based on weight/price, if it's free by voucher or if it's free because carrier has price equal to 0 |

### CODE REVIEW

File: src/Adapter/Cart/CartPresenter.php [hunk 1]
```diff
@@ -229,6 +229,50 @@ public function addCustomizedData(array $products, Cart $cart)
...
     }
 
+    protected function determineRealShippingStatusAndText($cart)
+    {
+        $configuration = Configuration::getMultiple(array(
+            'PS_SHIPPING_FREE_PRICE',
+            'PS_SHIPPING_FREE_WEIGHT'
+        ));
+
+        // Check if is real free shiping based on current cart
```
Comment (thierrymarianne) [inline src/Adapter/Cart/CartPresenter.php]: s/is real shiping/free shipping is actual/

**Comment by kpodemski** [comment]
If @vincentbz is willing to accept this PR I'll make a proper fix for this phrase

File: src/Adapter/Cart/CartPresenter.php [hunk 2]
```diff
@@ -229,6 +229,50 @@ public function addCustomizedData(array $products, Cart $cart)
...
+        $shippingCost = $cart->getTotalShippingCost(null, $this->includeTaxes());
+        $orderTotalwithDiscounts = $cart->getOrderTotal(true, Cart::BOTH_WITHOUT_SHIPPING, null, null, false);
+        $freeShippingFromPrice = (float) $configuration['PS_SHIPPING_FREE_PRICE'] > 0 && $orderTotalwithDiscounts >= (float) $configuration['PS_SHIPPING_FREE_PRICE']
+                                ? true
+                                : false
+        ;
```
Comment (julienbourdeau) [inline src/Adapter/Cart/CartPresenter.php]: I find this a bit hard to read. Usually ternary to return `true` or `false` are useless (even if there are many in the codebase).

This should be enough I think:
```php
$freeShippingFromPrice = (float) $configuration['PS_SHIPPING_FREE_PRICE'] > 0
                                             && $orderTotalwithDiscounts >= (float) $configuration['PS_SHIPPING_FREE_PRICE'];
```

Comment (kpodemski) [inline src/Adapter/Cart/CartPresenter.php]: you're right, i have more complex checking here and forgot to simplify it at the end, i'll work on details right after @vincentbz @xBorderie or whoever is in charge will give me green light with "

**Comment by xBorderie** [comment]
I'm not a PM, @vincentbz and @superGB are :)

**Comment by kpodemski** [comment]
Ok, so i'm waiting for @vincentbz i guess

**Comment by superGB** [comment]
Hi @kpodemski , with the same problem in mind, we already came up with another solution through the module ps_legalcompliance : http://forge.prestashop.com/browse/BOOM-1195, where we add "under conditions" if the default shipping cost is free. That way the software is compliant with german law.
free shipping under conditions
What's your opinion ?

**Comment by kpodemski** [comment]
@superGB not a bad idea but it would be good to:
- have possibility to disable this warning, for example if store sells only virtual products or have one really free shipping method
- disable this warning if we already have set `id_carrier` and `id_address_delivery` for cart because it means that "Free" is for real

I don't know, for me "To be determined" is beter than "Free" as we can have situation where we don't even offer "Free" shipping and shipping is still "0.00" at the beginning, what then? :)

**Comment by kpodemski** [comment]
Also, we can have situation where shipping is not free, it's paid, but it's paid based on `id_address_delivery`, as long as we don't have `id_address_delivery` and as long as customer doesn't really check carriers page we shouldn't show him any prices. 

From what i remember (from my customers feedback) - it was not only about "Free shipping"

**Comment by superGB** [comment]
Hi @kpodemski ,

> have possibility to disable this warning, for example if store sells only virtual products or have one really free shipping method
If there are only virtual products in your cart, the 'shipping' line won't be displayed anyway.

> disable this warning if we already have set id_carrier and id_address_delivery for cart because it means that "Free" is for real
I totally agree, we went for the easiest method.

From the moment your offer a choice of different carriers ou delivery countries, the shipping costs could always be false because you don't know what the address will be or which carrier the customer will choose.
I've never seen a web shop that displays 'to be determined' (or at least I don't remember). The common practice is to display a default shipping cost based on the customer's default country, with the default shipping method. Of course, the shipping cost displayed at this stage would still depends on the cart's content and on the cart promotion rules (free shipping starting from xx €). And maybe the merchant should be careful not to choose a default shipping method that is free (like pick up in store).

I agree with you that displaying "free" could be interpreted as "offered" and that's only true if you have set a cart or customer group promotion rule. If the default carrier has a price equal to 0, it should display 0.

In the end, I think the way it works today is ~OK except that we should provide the customer with feedback information about how the shipping cost is estimated at this stage. I've something in mind that could do the job.
So I'd argue to close this PR for the moment but if you're willing to discuss it, please use my email : guillaume.bruere@prestashop.com.

**Comment by kpodemski** [comment]
Well, I'm only providing you feedback from my customers, merchants. They don't want to show not real shipping costs, most of the time it was working fine with a text "Check costs" or "Tbd later" or whatever.

**Pull request closed without merging**